generators/csharp/sdk/src/test-generation/mock-server/BaseMockServerTestGenerator.ts

```diff
@@ -183,6 +205,47 @@ export class BaseMockServerTestGenerator extends FileGenerator<CSharpFile, SdkCu
         });
     }
 
+    protected generateMockAuthMethod(scheme: OAuthScheme): csharp.Method {
+        const tokenEndpointReference = scheme.configuration.tokenEndpoint.endpointReference;
+        const tokenEndpointHttpService = this.context.getHttpServiceOrThrow(tokenEndpointReference.serviceId);
+        const httpEndpoint = this.context.resolveEndpointOrThrow(
+            tokenEndpointHttpService,
+            tokenEndpointReference.endpointId
+        );
+        const allExamples = [...httpEndpoint.autogeneratedExamples, ...httpEndpoint.userSpecifiedExamples].map(
+            (example) => example.example
+        );
+        // TODO: support other response body types
+        const useableExamples = allExamples.filter((example): example is ExampleEndpointCall => {
+            const response = example?.response;
+            return response?.type === "ok" && response.value.type === "body";
+        });
+        useableExamples.forEach((example) => {
+            const jsonExample = example.request?.jsonExample as Record<string, unknown> | undefined;
+            if (!jsonExample) {
+                return;
+            }
+            for (const prop in jsonExample) {
+                let value = jsonExample[prop];
+                if (typeof value === "string") {
+                    value = value.toLowerCase();
+                    if (value === "client_id") {
+                        jsonExample[prop] = "CLIENT_ID";
+                    }
+                    if (value === "client_secret") {
+                        jsonExample[prop] = "CLIENT_SECRET";
+                    }
+                }
+            }
```

Comment:
Does this work work if the user defines this API with `camelCase`? This feels too specific / prone to fail. What are your thoughts?

Reply:
It'd be better if I can get the property name from IR, then use that to update the values in the examples.

Reply:
I updated the code to pull the path from IR and deep set it into the example (if the property is present in the example).